File: 75-sort-colors/sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let lo=0
    let hi=nums.length-1
    let k=0
    while(k<=hi){
        if(nums[k]===0){
            [nums[k],nums[lo]]=[nums[lo],nums[k]]
            lo=lo+1
            k=k+1
        }
        else if (nums[k]==2){
            [nums[hi],nums[k]]=[nums[k],nums[hi]]
            hi=hi-1
        }
        else{
            k=k+1
        }
    }
    
};

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zeroIndex=0
        arrayLength=len(nums)
        twoIndex=arrayLength-1
        index=0
        while index <=twoIndex:
            if nums[index]==2:
                nums[index],nums[twoIndex]=nums[twoIndex],nums[index]
                twoIndex=twoIndex-1
            elif nums[index]==0:
                nums[index],nums[zeroIndex]=nums[zeroIndex],nums[index]
                index=index+1
                zeroIndex=zeroIndex+1
            else:
                index=index+1
        return 
        